src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -272,7 +276,7 @@ def _normalize_properties(self, user, subscriptions, is_service_principal, cert_
             if subscription_dict[_SUBSCRIPTION_NAME] != _TENANT_LEVEL_ACCOUNT_NAME:
                 if hasattr(s, 'home_tenant_id'):
                     subscription_dict[_HOME_TENANT_ID] = s.home_tenant_id
-                if hasattr(s, 'managed_by_tenants'):
+                if hasattr(s, 'managed_by_tenants') and s.managed_by_tenants is not None:
                     subscription_dict[_MANAGED_BY_TENANTS] = [{_TENANT_ID: t.tenant_id} for t in s.managed_by_tenants]
```

Comment:
to invoid TypeError with trace like this:
_NoneType_ object is not iterable :
[_  File _usrlib64azlibpython3.6site-packagesknackcli.py_, line 215, in invoke\n    cmd_result = self.invocation.execute(args)\n_, _  File _usrlib64azlibpython3.6site-packagesazureclicorecommands__init__.py_, line 654, in execute\n    raise ex\n_, _  File _usrlib64azlibpython3.6site-packagesazureclicorecommands__init__.py_, line 718, in _run_jobs_serially\n    results.append(self._run_job(expanded_arg, cmd_copy))\n_, _  File _usrlib64azlibpython3.6site-packagesazureclicorecommands__init__.py_, line 711, in _run_job\n    six.reraise(*sys.exc_info())\n_, _  File _usrlib64azlibpython3.6site-packagessix.py_, line 703, in reraise\n    raise value\n_, _  File _usrlib64azlibpython3.6site-packagesazureclicorecommands__init__.py_, line 688, in _run_job\n    result = cmd_copy(params)\n_, _  File _usrlib64azlibpython3.6site-packagesazureclicorecommands__init__.py_, line 325, in __call__\n    return self.handler(*args, **kwargs)\n_, _  File _usrlib64azlibpython3.6site-packagesazureclicore__init__.py_, line 784, in default_command_handler\n    return op(**command_args)\n_, _  File _usrlib64azlibpython3.6site-packagesazureclicommand_modulesprofilecustom.py_, line 158, in login\n    use_cert_sn_issuer=use_cert_sn_issuer)\n_, _  File _usrlib64azlibpython3.6site-packagesazureclicore_profile.py_, line 240, in find_subscriptions_on_login\n    is_service_principal, bool(use_cert_sn_issuer))\n_, _  File _usrlib64azlibpython3.6site-packagesazureclicore_profile.py_, line 276, in _normalize_properties\n    subscription_dict[_MANAGED_BY_TENANTS] = [{_TENANT_ID: t.tenant_id} for t in s.managed_by_tenants]\n_]